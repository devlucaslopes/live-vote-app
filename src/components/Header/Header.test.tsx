import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'

import { Header } from '.'

const openNewSuggestion = jest.fn()

describe('<Header />', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Header openNewSuggestion={openNewSuggestion}>Lorem ipsum</Header>
    )

    expect(screen.getByRole('heading').textContent).toBe('Lorem ipsum')
    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should call openNewSuggestion() when button is clicked', () => {
    render(<Header openNewSuggestion={openNewSuggestion}>Lorem ipsum</Header>)

    fireEvent.click(screen.getByRole('button'))

    expect(openNewSuggestion).toHaveBeenCalledTimes(1)
  })
})
