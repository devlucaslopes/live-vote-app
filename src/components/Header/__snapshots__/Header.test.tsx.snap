// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<div>
  <div
    class="flex justify-between"
  >
    <h1
      class="text-4xl text-white"
    >
      Lorem ipsum
    </h1>
    <button
      class="flex items-center text-pink-500"
      type="button"
    >
      <svg
        class="mr-2"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="12"
          x2="12"
          y1="5"
          y2="19"
        />
        <line
          x1="5"
          x2="19"
          y1="12"
          y2="12"
        />
      </svg>
      Nova sugest√£o
    </button>
  </div>
</div>
`;
